### PR TITLE
Add sqlproj property to trace the origin of the project.

### LINKED ISSUE #15941: Add project property to hold the source database information in sql-database-project extension
When migration  database to Azure Data we would need to keep track of origin databases for telemetry purposes. This would help us know how many databases are actually deployed to SQL/ Azure SQL.
In current solutions we use the "Application Name" connection string property to identify on the target the deployed databases. In ADS we do not own the deployment process and "Application Name" would not seem reasonable for sql-database-project extension.

 @nahk-ivanov  proposal is to have a <DatabaseSource/> project property which could be set/updated  and include it in the telemetry.  This would allow us to track at least something similar to what we’ve had. It  can be also used for “export database”, i.e. <DatabaseSource>extract-schema</DatabaseSource>, <DatabaseSource>extension-name</DatabaseSource> or multiple <DatabaseSource>extract-schema;extension-name</DatabaseSource>.

### PR DESCRIPTION
**Pull request by nahk-ivanov**
As part of the database migration process (schema conversion, in particular) we want to be able to tell when converted schemas are being built/deployed to the actual database server. Given that we rely on the SQL Database Projects ADS extension for the compilation/deployment, we don't have too many options other than updating the said extension.

The suggested approach is to make the following changes:
1) Add new property to the sqlproj file (called `DatabaseSource`), which will maintain the origin(s) of the project. The property can contain multiple values (separated by semicolon), in case same project contains objects produced by multiple sources (extract schema, convert from another database, etc.).
2) During build and deploy actions, send the well-known values from the newly added property to the telemetry. We don't want to send any random value of the property, as it may raise some privacy concerns. Instead we define a list of the well-known values that we know do not carry any personal information and send those, if they are specified.

This change adds all necessary APIs to the SQl Database projects extension which will be consumed by our migration extensions to populate new `DatabaseSource` property.

This PR fixes #15941

### CODE REVIEW

**Comment by nahk-ivanov** [comment]
I took over the change from Steven (#17781). Most of the heavy lifting was already done, I just slightly updated the code to address the comment. There were few points that were still open in the previous PR, so I updated them according to the requests from the owners (even though I personally don't agree with them) to speed up the process.

File: extensions/sql-database-projects/src/models/project.ts
Comment (kisantia) [inline extensions/sql-database-projects/src/models/project.ts:1756]: use undefined instead of null since we try to use undefined instead of null throughout the code base

**Review by kisantia — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
@kisantia Do we still need this anymore?

**Comment by kisantia** [comment]
@smartura does your team still needs this change?

**Comment by smartura** [comment]
@kisantia I'm no longer on this team, @benrr101 is this still needed?

**Comment by benrr101** [comment]
I don't know the intention of this change. Let me sync with Steve offline and see if I can get any idea of why this was on the docket.

**Comment by nahk-ivanov** [comment]
@benrr101 We want to know how many of our converted schemas actually get deployed (or at least built into dacpacs). Since that happens outside of our extensions, we have to collect it from "SQL Database Projects" extension. The problem is that when "SQL Database Projects" extension runs build or publish - it doesn't know where the scripts came from. So our "contract" is not add additional project property (`DatabaseSources`) which will hold the information about the origins of the scripts. This can later be used by "extract database schema" feature as well, etc.

**Comment by benrr101** [comment]
@kisantia Hi Kim, after chatting with Steve and with the input from Alex, it looks like we will still want to get this change put into the SQL db projects. We don't know exactly when we'll start utilizing it, but having that mechanism available to use will be useful when we start looking into telemetry again. If there's any changes still required to be made, please let me know.

**Comment by kisantia** [comment]
@benrr101 ok you can go ahead and merge this if there are still plans to use this. Please merge from main and verify tests are passing before merging this PR.

**Comment by benrr101** [comment]
@kisantia I'm not a contributor to this project, so I can't merge it myself. There's no conflicts, so you should be good to go to push the merge button :)

**Pull request merged**
